src/lib/Imp.hs

```diff
@@ -107,9 +114,23 @@ translateTopLevel (maybeDest, block) = do
   return (reconAtom, resultIExprs)
 
 runImpM :: ImpCtx -> Scope -> ImpM a -> a
-runImpM opts inVarScope m =
+runImpM ctx inVarScope m =
   fromRight (error "Unexpected top level error") $
-    fst $ runCat (runReaderT (runExceptT m) opts) $ mempty {envScope = inVarScope}
+    fst $ runCat (runReaderT (runExceptT m) ctx) $ mempty {envScope = inVarScope}
+
+toImpStandalone :: Name -> Atom -> ImpM ImpFunction
+toImpStandalone fname ~(LamVal b body) = do
+  let argTy = binderAnn b
+  let outTy = getType body
+  -- TODO(dougalm): need to think carefully about these pointers!
+  (ptrSizes, ~(Con (ConRef (PairCon outDest argDest)))) <- fromEmbed $
+    makeDest (LLVM, CPU, Unmanaged) (PairTy outTy argTy)
```

Comment:
Hmmm this is a bit shady. What if we're using the GPU backend?

Reply:
Yes it is shady. If we're using the GPU backend then we'll need to specialize two versions of any non-inlined function - one for GPU and one for CPU. I'll add it to the follow-up todos to track it and put an assert here for now.